r1cs/gadgets/std/src/bits/uint32.rs

```diff
@@ -133,6 +163,94 @@ impl UInt32 {
         }
     }
 
+    pub fn shr(&self, by: usize) -> Self {
+        let by = by % 32;
+
+        let fill = Boolean::constant(false);
+
+        let new_bits = self
+            .bits
+            .iter() // The bits are least significant first
+            .skip(by) // Skip the bits that will be lost during the shift
+            .chain(Some(&fill).into_iter().cycle()) // Rest will be zeros
+            .take(32) // Only 32 bits needed!
+            .cloned()
+            .collect();
+
+        UInt32 {
+            bits: new_bits,
+            value: self.value.map(|v| v >> by as u32),
+        }
+    }
+
+    fn triop<ConstraintF, CS, F, U>(
```

Comment:
Same comment as in Boolean: please let's move these functions to the sha256 related file

Reply:
Done